grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details.ts

```diff
@@ -15,58 +15,63 @@ import { isNonNull } from '@app/lib/preconditions';
-
+export class OsqueryDetails extends Plugin {
   readonly flowRunning$ = this.flagByState(FlowState.RUNNING);
   readonly flowCompleted$ = this.flagByState(FlowState.FINISHED);
   readonly flowError$ = this.flagByState(FlowState.ERROR);
 
   readonly args$: Observable<OsqueryArgs> = this.flowListEntry$.pipe(
-      map((flowListEntry) => flowListEntry.flow.args as OsqueryArgs)
+      map(flowListEntry => flowListEntry.flow.args as OsqueryArgs),
   );
 
-  readonly osqueryResult$: Observable<OsqueryResult> = this.flowListEntry$.pipe(
-    flatMap((listEntry) => listEntry.resultSets),
-    flatMap((singleResultSet) => singleResultSet?.items),
+  readonly osqueryResults$: Observable<OsqueryResult> = this.flowListEntry$.pipe(
+    flatMap(listEntry => listEntry.resultSets),
+    flatMap(singleResultSet => singleResultSet?.items),
     filter(isNonNull),
-    map((singleItem) => singleItem.payload as OsqueryResult),
+    map(singleItem => singleItem.payload as OsqueryResult),
+  );
+
+  readonly resultsTable$ = this.osqueryResults$.pipe(
+    map(result => result?.table),
   );
 
-  readonly resultTable$ = this.osqueryResult$.pipe(
-    map((result) => result?.table),
+  readonly resultsStderr$ = this.osqueryResults$.pipe(
+    map(result => result?.stderr),
   );
 
-  readonly resultStderr$ = this.osqueryResult$.pipe(
-    map((result) => result?.stderr)
+  readonly osqueryProgress$: Observable<OsqueryProgress> = this.flowListEntry$.pipe(
+    map(listEntry => listEntry.flow.progress as OsqueryProgress),
+    filter(isNonNull),
+  );
+
+  readonly progressTable$ = this.osqueryProgress$.pipe(
+    map(progress => progress.partialTable),
+  );
+
+  readonly additionalRowsAvailable$ = this.osqueryProgress$.pipe(
+    map(progress => {
+      const progressTableLength = progress.partialTable?.rows?.length;
+      const fullTableLength = progress.totalRowsCount;
+
+      if (isNonNull(progressTableLength) && isNonNull(fullTableLength)) {
+        return BigInt(fullTableLength) - BigInt(progressTableLength);
+      } else {
+        return null;
+      }
+    }),
   );
 
   private flagByState(targetState: FlowState): Observable<boolean> {
     return this.flowListEntry$.pipe(
-      map((listEntry) => listEntry.flow.state === targetState)
+      map(listEntry => listEntry.flow.state === targetState)
     );
   }
 
-  constructor() {
-    super();
-    this.flowCompleted$
-      .pipe(
-        takeUntil(this.unsubscribe$),
-        filter(completed => completed),
-        take(1))
-      .subscribe(() => this.loadResults());
-  }
-
-  private loadResults() {
-    this.queryFlowResults({offset: 0, count: 1});
-  }
-
-  ngOnDestroy() {
-    this.unsubscribe$.next();
-    this.unsubscribe$.complete();
+  private loadCompleteResults() {
+    this.queryFlowResults({offset: 0, count: 1}); // TODO: Fetch more chunks if present
   }
 
-  trackByIndex(index: number, item: unknown): number {
-    return index;
+  fullTableRequested() {
```

Comment:
Use imperative over past tense, you could also call `loadCompleteResults()` directly?